syntax = "proto3";

// Client -> Server
message HandshakeHello {
  bytes kem_public_key = 1;   // Kyber-512 pk
  uint32 version       = 2;   // 1
  bytes identity_pub   = 3;   // Ed25519 long-term public key (32 bytes)
  bytes identity_sig   = 4;   // Sig over context || kem_public_key
}

// Server -> Client
message HandshakeResponse {
  bytes kem_ciphertext = 1;   // Kyber-512 ct
  uint32 version       = 2;   // 1
  bytes identity_pub   = 3;   // Server Ed25519 public key
  bytes identity_sig   = 4;   // Sig over context || kem_ciphertext || client_kem_public_key
}
